support/controlplane-component/status.go

```diff
@@ -183,14 +186,64 @@ func (c *controlPlaneWorkload[T]) setRolloutCompleteCondition(cpContext ControlP
 			Reason:  string(apierrors.ReasonForError(err)),
 			Message: err.Error(),
 		})
-		return
+		return metav1.ConditionFalse
+	}
+
+	checkOperandsRolloutStatusFn := c.checkOperandsRolloutStatus
+	if c.customOperandsRolloutCheck != nil {
+		checkOperandsRolloutStatusFn = c.customOperandsRolloutCheck
 	}
 
 	status, reason, message := c.workloadProvider.IsReady(workloadObject)
+
+	operandsReady, err := checkOperandsRolloutStatusFn(cpContext.workloadContext())
+	if !operandsReady && status == metav1.ConditionTrue {
+		status = metav1.ConditionFalse
+		reason = "WaitingForOperands"
+		message = err.Error()
+	}
+
 	meta.SetStatusCondition(conditions, metav1.Condition{
 		Type:    string(hyperv1.ControlPlaneComponentRolloutComplete),
 		Status:  status,
 		Reason:  reason,
 		Message: message,
 	})
+	return status
+}
+
+func (c *controlPlaneWorkload[T]) checkOperandsRolloutStatus(cpContext WorkloadContext) (bool, error) {
```

Comment:
we've got code that does a similar thing in the hypershift install cli subcommand, I wonder if  you can reuse?
https://github.com/openshift/hypershift/blob/main/cmd/install/install.go#L405

Reply:
I don't see how it's similar, this is a one shot check that looks for specific label/annotation. Whereas the cli code waits until the deployment is ready.
fwiw the CLI code should be using `util.IsDeploymentReady()`

Reply:
The similarity I was referring to is around the high level functionality ― checking a rollout status of a deployment. 

I think what you pointed out is exactly what I was looking for, maybe we can in future refactor the CLI code to reuse `util.IsDeploymentReady()`. Thanks!